### PR TITLE
Upgrade solvation thermo GAV and solute & solvent libraries

### PR DESCRIPTION
**Pull request by yunsiechung**
_**Note that this pull request requires simultaneous updates of RMG-Py and RMG-database. The pull request on RMG-Py solvation_thermo_GAV branch is https://github.com/ReactionMechanismGenerator/RMG-Py/pull/2112. The last commit needs to be removed before merging this branch.**_

### Motivation or Problem
The previous group additivity (GAV) method for solvation thermo uses very limited number of groups. This pull request will upgrade the solvation thermo GAV method that uses a lot more groups. Also, this pull request adds a lot more solvents and solutes to the solvent and solute libraries. 

### Description of Changes
The major changes include:

1. The solvation GAV now follows an approach similar to the gas-phase thermo GAV and uses `group`, `ring`, `polycyclic`, `longDistanceInteraction_cyclic`, `longDistanceInteraction_noncyclic`, `halogen`, and `radical` corrections to estimate solute parameters (E, S, A, B, L). All the solvation groups are fitted by me (more detailed reference will be added to the RMG-database once this work get published). Each solvation group includes a new attribute `dataCount` stored as either a `DataCountGAV` class or `None`. This new attribute stores the number of experimental solute parameter data used to fit each group value.

2. Many more solutes and solvents are added to the solute and solvent libraries. The new solvent parameters are fitted by me (more detailed reference will be added to the RMG-database once this work get published), and the number of solvation free energy and enthalpy data used to fit these parameters are included in the solvent library entries. The solvation free energy and enthalpy errors associated with the fitted solvent parameters are also included.

### CODE REVIEW

File: input/thermo/libraries/primaryThermoLibrary.py [hunk 1]
```diff
@@ -1136,3 +1136,39 @@
 level of theory: CCSD(T)F12A/cc-pVTZ-F12//B3LYP/6-311++g(d,p) + BAC
 """,
 )
+
+entry(
+    index = 46,
+    label = "OCCCO",
```
Comment (mjohnson541) [inline input/thermo/libraries/primaryThermoLibrary.py:1152]: Should we have similar molecules such as  O=C=C=O as well in primaryThermoLibrary?

File: input/thermo/libraries/primaryThermoLibrary.py [hunk 2]
```diff
@@ -903,13 +903,17 @@
         Cp0 = (29.1007,'J/mol/K'),
         CpInf = (37.4151,'J/mol/K'),
     ),
-    shortDesc = u"""71STUPRO""",
+    shortDesc = u"""71STUPRO JANAF 2nd ed. (1971)""",
```
Comment (mjohnson541) [inline input/thermo/libraries/primaryThermoLibrary.py:906]: Do we have an idea how accurate these are?

File: input/solvation/libraries/solute.py [hunk 3]
```diff
@@ -4,8 +4,9 @@
 name = "Solute Descriptors"
 shortDesc = u""
 longDesc = u"""
-From Abraham J. Chem. Soc. 1994
+From Abraham J. Chem. Soc. 1994 or from in-house database from Prof. Abraham
```
Comment (mjohnson541) [inline input/solvation/libraries/solute.py:7]: In both of these libraries (this commit and the one before, and for the groups files as well (although those are likely to be more homogeneous)) there are lots of entries without long descriptions. Can you say something about the default source of values at the top of the file and add a description for the other ones?

**Review by mjohnson541 — COMMENTED** [review]
Thanks! This needs to pass unit tests, but the commits look mostly okay. Lots of RMG-tests changes in the solvation example do the changes look reasonable to you?

File: input/thermo/libraries/primaryThermoLibrary.py [hunk 4]
```diff
@@ -884,3 +884,255 @@
...
+            NASAPolynomial(coeffs=[3.43657,0.000486021,-1.2524e-06,1.36475e-09,-4.09574e-13,-33800.1,1.20682], Tmin=(298,'K'), Tmax=(1250,'K')),
+            NASAPolynomial(coeffs=[2.7813,0.00103959,-2.41735e-07,2.68416e-11,-1.09766e-15,-33504.2,5.0197], Tmin=(1250,'K'), Tmax=(3000,'K')),
+        ],
+        Tmin = (298,'K'),
+        Tmax = (3000,'K'),
+        E0 = (-281.113,'kJ/mol'),
+        Cp0 = (29.1007,'J/mol/K'),
+        CpInf = (37.4151,'J/mol/K'),
+    ),
+    shortDesc = u"""71STUPRO""",
```
[the remark this answers is not in the document]
Comment (yunsiechung) [reply input/thermo/libraries/primaryThermoLibrary.py]: This is not a commit added by me, so I will close this. I think the previous version was not rebased properly with the official/master branch and somehow this PR showed other people's commits as well. I rebased again and pushed again.

File: input/thermo/libraries/primaryThermoLibrary.py (hunk 1 above)
Comment (yunsiechung) [reply input/thermo/libraries/primaryThermoLibrary.py]: This is not a commit added by me, so I will close this. I think the previous version was not rebased properly with the official/master branch and somehow this PR showed other people's commits as well. I rebased again and pushed again.

File: input/thermo/libraries/primaryThermoLibrary.py (hunk 2 above)
Comment (yunsiechung) [reply input/thermo/libraries/primaryThermoLibrary.py]: This is not a commit added by me, so I will close this. I think the previous version was not rebased properly with the official/master branch and somehow this PR showed other people's commits as well. I rebased again and pushed again.

File: input/solvation/libraries/solute.py (hunk 3 above)
Comment (yunsiechung) [reply input/solvation/libraries/solute.py]: I added the reference in the long description of each entry for `solvent.py` and `solute.py` (solvent and solute libraries), which are done in the fixup commits 4fa1bb3 and 4b83d36.
I added a bit more detailed explanation in the long description at the top of the file for solvation group files in the fixup commit ebdaf4c.

**Comment by yunsiechung** [comment]
@mjohnson541 I made all the changes you requested. You can see my comment for each of your requested change.

If everything looks good for both this and its twin RMG-Py PR  [#2112](https://github.com/ReactionMechanismGenerator/RMG-Py/pull/2112), I can squash the fixup commits, remove the temporary commit for simultaneous updates of RMG-Py and RMG-database (30f35ac), and force push.

**Review by mjohnson541 — APPROVED** [review]
Looks good to squash and then merge to me!

**Comment by yunsiechung** [comment]
@mjohnson541 I squashed and rebased. But I did not remove the temporary commit (0155ed3) just to make sure this passes all the tests.
When you merge, please remove the temporary commit (0155ed3) for both this and RMG-Py PR.

**Pull request merged**